### DISCUSSION TITLE
Git Status Idea

### DISCUSSION THREAD

**Discussion by philipbordallo** [starship/starship#1372] [General]
I've been using **starship** for a while now and it's been great. But one thing I've noticed is the git status symbols are a bit overwhelming. There is a lot of them, and their meaning and reason for showing up isn't always clear.

While I was looking to customize them/clean things up I ran into #1371 and it made me realize is there a benefit to having that level of detail? Do I care whether something was modified, deleted, or renamed as well as all the other information? Or is it more important to know if something has changed, is untracked, has a conflict, is staged, has been stashed or is in-sync with the remote?

## Idea

- Add a `changed` state to handle all modifications and deletions in the working directory.
- Update `staged` state to included modifications, deletions and renames that have been staged.
- Remove `deleted` and `renamed` states.

This would mean that the states for git status would be:

state | description 
--- | ---
`changed` | A file has been modified or deleted in the working directory.
`untracked` | A new file that is not been tracked has been added.
`conflicted` | There is a conflict in the current branch.
`staged` | A file that has been modified, deleted, or renamed has been staged.
`stashed` | A stash exists for the local repository.
`ahead` |	This branch is ahead of the branch being tracked.
`behind` |	This branch is behind of the branch being tracked.
`diverged` |	This branch has diverged from the branch being tracked.

### Examples

Given these files:
```
.
├── one.txt
└── two.txt
```
And these sequential actions taken:
1. **one.txt** has been deleted, the state would be `changed`.
1. **two.txt** has been renamed to **three.txt**, the state would be `changed` and `untracked`.
1. The deleted **one.txt** has been staged, the state would be `changed`, `staged`, and `untracked`
1. All the other changes have been staged, the state would be `staged`

This way at a quick glance you can see what state git is in and if more information needs to be gleaned (did these changes include a deletion or was it a rename?) running `git status` can give you more details.

**Comment by matchai** [comment]
Thanks for the suggestion Philip! 😄 

I can definitely see how the git status module can get noisy and overwhelming. There's certainly room to simplify the git status module to show less details, and for what it's worth, Starship can be _considerably_ faster if we could stop checking for git status as soon as a single file was in one of these states. 🚀
